tests/has/devfile_private_source.go

```diff
@@ -0,0 +1,178 @@
+	Expect(err).NotTo(HaveOccurred())
+
+	// Wait for the resource to be in the "AwaitingTokenData" phase
+	var spiAccessTokenBinding *v1beta1.SPIAccessTokenBinding
+	Eventually(func() bool {
+		// application info should be stored even after deleting the application in application variable
+		spiAccessTokenBinding, err = framework.SPIController.GetSPIAccessTokenBinding(spiAccessTokenBindingName, namespace)
+		if err != nil {
+			return false
+		}
+		return spiAccessTokenBinding.Status.Phase == v1beta1.SPIAccessTokenBindingPhaseInjected || (spiAccessTokenBinding.Status.Phase == v1beta1.SPIAccessTokenBindingPhaseAwaitingTokenData && spiAccessTokenBinding.Status.OAuthUrl != "")
+	}, 1*time.Minute, 100*time.Millisecond).Should(BeTrue(), "SPI controller didn't set SPIAccessTokenBinding to AwaitingTokenData/Injected")
+
+	if spiAccessTokenBinding.Status.Phase == v1beta1.SPIAccessTokenBindingPhaseAwaitingTokenData {
+		// If the phase is AwaitingTokenData then manually inject the git token
+		// Get the oauth url and linkedAccessTokenName from the spiaccesstokenbinding resource
+		oauthURL := spiAccessTokenBinding.Status.OAuthUrl
+		parsedOAuthURL, err := url.Parse(oauthURL)
+		Expect(err).NotTo(HaveOccurred())
+		oauthHost := parsedOAuthURL.Host
+		linkedAccessTokenName := spiAccessTokenBinding.Status.LinkedAccessTokenName
+
+		// Before injecting the token, validate that the linkedaccesstoken resource exists, otherwise injecting will return a 404 error code
+		Eventually(func() bool {
+			// application info should be stored even after deleting the application in application variable
+			_, err := framework.SPIController.GetSPIAccessToken(linkedAccessTokenName, namespace)
+			return err == nil
+		}, 1*time.Minute, 100*time.Millisecond).Should(BeTrue(), "SPI controller didn't create the SPIAccessToken")
+
+		// Now that the spiaccesstokenbinding is in the AwaitingTokenData phase, inject the GitHub token
+		var bearer = "Bearer " + string(bearerToken)
+		var jsonStr = []byte(`{"access_token":"` + token + `"}`)
+		http.DefaultTransport.(*http.Transport).TLSClientConfig = &tls.Config{InsecureSkipVerify: true}
+		req, err := http.NewRequest("POST", "https://"+oauthHost+"/token/"+namespace+"/"+linkedAccessTokenName, bytes.NewBuffer(jsonStr))
+		Expect(err).NotTo(HaveOccurred())
+		req.Header.Add("Authorization", bearer)
+		req.Header.Set("Content-Type", "application/json")
+
+		client := &http.Client{}
+		resp, err := client.Do(req)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(resp.StatusCode).Should(Equal(204))
+		defer resp.Body.Close()
+
+		// Check to see if the token was successfully injected
+		Eventually(func() bool {
+			// application info should be stored even after deleting the application in application variable
+			spiAccessTokenBinding, err = framework.SPIController.GetSPIAccessTokenBinding(RedHatAppStudioApplicationName, namespace)
```

Comment:
Should we use SPIAccessTokenBindingName variable here?

Reply:
Yup, good catch. That's been fixed 🙂